Change serve package to live-server for refresh on save

citypaul opened the pull request:
Using this package creates a nicer user experience - at least when I tried it, the `serve` package wasn't reloading on save. Using this package should solve that problem :-)

joshwcomeau commented:
Yeah, so before `serve` we were using Parcel, and it added all sorts of friction around Node versions and incompatible peer deps and all the fun of the NPM ecosystem 🙃

Looking at live-server, it worries me a little bit that there isn't a lockfile (`yarn.lock` or `package-lock.json`), and that the dependency versions are loose (eg. `^3.0.4` instead of `3.0.4`). I'm sure everything works great right now but I worry that it can break any day when a dependency pushes an incompatible minor/patch version.

That said, I do recognize that `serve` is pretty minimal, and it's not as nice as something that auto-refreshes… Let me give it some thought.

citypaul commented:
Hey @joshwcomeau!

Thanks for your comments. I've added a package-lock here for you. Hopefully that helps!

citypaul commented:
@joshwcomeau I've made the above change here, so this will currently restrict users to only be able to use `yarn`. If they try to use npm, they will get this message:

![image](https://user-images.githubusercontent.com/5850178/118844718-18415380-b8c3-11eb-94de-7941962325cd.png)

I also deleted the package-lock and updated the yarn.lock instead.

Try checking out the branch and doing an `npm install` :)

citypaul commented on package.json:
```diff
@@ -4,9 +4,13 @@
+    "live-server": "^1.2.1"
+  },
+  "engines": {
+    "npm": "please-use-yarn",
```
This change will prevent `npm install` from working (it has to be combined with the `engine-strict = true` setting in the `.npmrc`

joshwcomeau commented:
So after a Twitter poll and asking around, it seems like NPM is more widely-used, and is becoming more popular (I think Yarn 2's wildly-different approach has scared some folks away from Yarn).

I integrated your suggestion (to use `live-server`) in the [latest commit](https://github.com/css-for-js/huckleberry/commit/fdef9f9b9ba9f1a9c26e85e6ef3cf991e24dd305), and I also removed the yarn lockfile and replaced it with `package-lock`. I thought about trying to restrict folks from using Yarn, but figured that was a bit too aggressive. Might be worth revisiting if we get a bunch of folks having issues because Yarn pulls in different dependencies, but I wanna wait and see.

Thanks for your work on this, and sorry I didn't merge it directly! I credited you in the latest commit's message.
